From fe68a5de3c47a233fc971ddd2c8af7c6c578c8cb Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Thu, 11 Aug 2022 14:14:29 -0400
Subject: [PATCH] qemu: vcpu: avoid using ifdef CONFIG_KVM in common code.

CONFIG_KVM should not be used in common code.  If we look at:

commit cbca3722a33c ("include/exec/poison: Mark CONFIG_KVM as poisoned, too")
we see that it states:

    CONFIG_KVM is only defined for target-specific code, so nobody should
    use it by accident in common code. To avoid such subtle bugs,
    CONFIG_KVM is now marked as poisoned in common code.

Since there already is a kvm-stub for when CONFIG_KVM is not enabled,
we can make use of that and make the common code less "ifdeffy" at
the same time.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>

Rebase to v7.2.0
Signed-off-by: He Zhe <zhe.he@windriver.com>

---
 accel/stubs/kvm-stub.c | 5 +++++
 softmmu/vl.c           | 2 --
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/accel/stubs/kvm-stub.c b/accel/stubs/kvm-stub.c
index 5d2dd8f35..24ca15b5a 100644
--- a/accel/stubs/kvm-stub.c
+++ b/accel/stubs/kvm-stub.c
@@ -12,6 +12,7 @@
 
 #include "qemu/osdep.h"
 #include "sysemu/kvm.h"
+#include "sysemu/cpus.h"
 #include "hw/pci/msi.h"
 
 KVMState *kvm_state;
@@ -132,3 +133,7 @@ uint32_t kvm_dirty_ring_size(void)
 {
     return 0;
 }
+
+void vcpu_parse(const char *optarg)
+{
+}
diff --git a/softmmu/vl.c b/softmmu/vl.c
index b112e9e39..e8bd3268f 100644
--- a/softmmu/vl.c
+++ b/softmmu/vl.c
@@ -3253,9 +3253,7 @@ void qemu_init(int argc, char **argv)
                                            "smp", optarg);
                 break;
             case QEMU_OPTION_vcpu:
-#ifdef CONFIG_KVM
                 vcpu_parse(optarg);
-#endif
                 break;
             case QEMU_OPTION_vnc:
                 vnc_parse(optarg);
